lib/dry/system/identifier.rb

```diff
@@ -0,0 +1,168 @@
+# frozen_string_literal: true
+
+require "dry-equalizer"
+require_relative "constants"
+
+module Dry
+  module System
+    # An identifier representing a component to be registered.
+    #
+    # Components are eventually registered in the container using plain string
+    # identifiers, available as the `identifier` or `key` attribute here. Additional
+    # methods are provided to make it easier to evaluate or manipulate these identifiers.
+    #
+    # @api public
+    class Identifier
+      include Dry::Equalizer(:identifier, :namespace, :separator)
+
+      # @!attribute [r] identifier
```

Comment:
This macro is not needed, YARD detects attr_reader syntax so a standard method tags will work here

Reply:
Ah cool, thanks. I just copied those from elsewhere in the code. I'll adjust!